File: 01a.exs
IO.read(:all)
  |> String.strip
  |> String.split(", ")
  |> Enum.map(fn command -> String.split_at(command, 1) end)
  |> Enum.map(fn {dir, len} -> {dir == "L", String.to_integer(len)} end)
  |> Enum.scan({0, -1, nil}, fn ({left, len}, {fx, fy, _}) -> if left do {fy, -fx, len} else {-fy, fx, len} end end)
  |> Enum.reduce({0, 0}, fn ({fx, fy, len}, {x, y}) -> {x + len * fx, y + len * fy} end)
  |> (fn {x, y} -> abs(x) + abs(y) end).()
  |> IO.inspect

File: 01a.rb
puts gets
  .strip
  .split(", ")
  .map { |command| [command[0], command[1..-1].to_i] }
  .inject([0, -1, 0, 0]) { |(dx, dy, x, y), (dir, len)| dir == 'L' ? [dy, -dx, x + len * dy, y + len * -dx] : [-dy, dx, x + len * -dy, y + len * dx] }[2..3]
  .map(&:abs)
  .inject(:+)
